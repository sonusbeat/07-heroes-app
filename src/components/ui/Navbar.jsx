import { Link, NavLink } from "react-router-dom";

const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <Link className="navbar-brand" to="/">
        Asociaciones
      </Link>

      <div className="navbar-collapse mr-auto">
        <div className="navbar-nav">
          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/marvel"
          >
            Marvel
          </NavLink>

          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/dc"
          >
            DC
          </NavLink>

          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/hero-search"
          >
            Search
          </NavLink>
        </div>{/* navbar-nav */}
      </div>{/* navbar-collapse */}

      <div className="navbar-collapse ml-auto">
        <div className="navbar-nav">
          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/login"
          >
            Logout
          </NavLink>
        </div>{/* navbar-nav */}
      </div>{/* navbar-collapse */}

    </nav>
  );
};

export default Navbar;