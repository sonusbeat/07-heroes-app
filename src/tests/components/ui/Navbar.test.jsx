import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router';
import AuthContext from '../../../auth/AuthContext';
import Navbar from '../../../components/ui/Navbar';
import types from '../../../types/types';

describe("Pruebas en el componente <Navbar />", () => {
  const historyMock = {
    push: jest.fn(),
    replace: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Daniel"
    }
  };

  afterEach(() => {
    jest.clearAllMocks()
  });

  const wrapper = mount(
    <AuthContext.Provider value={ contextValue }>
      <MemoryRouter>
        <Router history={ historyMock }>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test('Debe coincidir con el Snapshot', () => {

    expect( wrapper ).toMatchSnapshot();

  });

  test('Debe de mostrarse el nombre de usuario en el menu', () => {

    expect( wrapper.find(".text-info").text().trim() ).toBe( contextValue.user.name );

  });


  test('Debe de llamar el logout y usar el history', () => {

    wrapper.find("button").prop("onClick")();

    expect( contextValue.dispatch ).toHaveBeenCalledWith({
      type: types.logout
    });

    expect( historyMock.replace ).toHaveBeenCalledWith("/login");

  });

});
