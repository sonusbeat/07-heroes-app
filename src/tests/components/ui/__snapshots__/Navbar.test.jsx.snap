// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Navbar /> Debe coincidir con el Snapshot 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse mr-auto"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse ml-auto"
  >
    <div
      className="navbar-nav"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Daniel
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</nav>
`;
